src/store/slices/animeMangaSlice.ts

```diff
@@ -0,0 +1,31 @@
+"use client";
+import { createSlice } from "@reduxjs/toolkit";
+import { AnimeManga } from "../../types";
+
+type ContentState = {
+  content: AnimeManga[];
+  loading: boolean;
+};
+
+const initialState: ContentState = {
+  content: [],
+  loading: true,
+};
+
+const animeMangaSlice = createSlice({
+  name: "anime",
+  initialState,
+  reducers: {
+    setContent: (state, action) => {
+      state.content = action.payload;
+      state.loading = false;
+    },
+
+    setLoading: (state) => {
+      state.loading = true;
+    },
```

Comment:
Instead of doing this, you can receive in the setLoading reducer a prop indicating if you want to set it as true or false then you do in your components

```
dispatch(setLoading(true))


dispatch(setLoading(false))
```